build/testing/sauce-labs/test-browsers.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * @name 
+ * Test Browsers
+ * 
+ * @description 
+ * Cross browser testing with Selenium Webdriver (browser configurations) and Appium 
+ * (device configurations) for testing on Sauce Labs. The configurations are sourced
+ * from 'build/testing/config-browsers.js'.
+ * 
+ * @example 
+ * To execute on CLI, run 'node build/testing/sauce-labs/test-browsers.js' from root directory.
+ * 
+ * @requires
+ * You must create environment variables if executing this script locally using CLI. This can be done 
+ * by adding to your '~/.bashrc' file.
+ * 
+ * export SAUCE_LABS_USER=[some value]
+ * export SAUCE_LABS_USER=[some value] 
+ *
+ * CircleCI has these values included as environment variables
+ * 
+ * @see
+ * To understand in detail:
+ * Getting Started: https://help.crossbrowsertesting.com/selenium-testing/getting-started/javascript/
+ * Scripting docs: https://www.seleniumhq.org/docs/03_webdriver.jsp#chapter03-reference 
+ */
+const { Builder } = require("selenium-webdriver");
+const chalk = require('chalk');
+const { Configure, Phases } = require("../config-browsers.js");
+const { spawn } = require('child_process');
+const git = spawn('git', ['rev-parse', '--abbrev-ref', 'HEAD']);

```

Comment:
for readability, I would consider moving this away from the modules that are being required in.

Reply:
Could you clarify what is "this"?  Are you talking about just the git / spawn line?

Reply:
I moved near the rest of the Git work lower in the file.